### PR TITLE
Improving DX for iOS setup

### PR DESCRIPTION
**Pull request by StuartHarris**
As a step towards improving the developer experience whilst getting started, this PR updates the docs on how to set up an iOS project (for use with Crux) using [XcodeGen](https://github.com/yonaskolb/XcodeGen).

### CODE REVIEW

File: docs/src/getting_started/ios_with_xcodegen.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+{{#include ../../../examples/hello_world/iOS/CounterApp/ContentView.swift}}
+```
+
+And create `iOS/CounterApp/CounterAppApp.swift` to look like this:
+
+```swift
+{{#include ../../../examples/hello_world/iOS/CounterApp/CounterAppApp.swift}}
+```
+
+Run `xcodegen` again to update the Xcode project with these newly created source files (or add them manually in Xcode to the `CounterApp` group), and then open `iOS/CounterApp/CounterApp.xcodeproj` in Xcode. You might need to select the `CounterApp` scheme, and an appropriate simulator, in the drop-down at the top, before you build.
```
Comment (charypar) [inline docs/src/getting_started/ios_with_xcodegen.md:83]: What's the intended workflow with xcodegen - is it a "generate once, then use the project", or is it a "yaml file is the source of truth and the xcode project gets regenerated all the time"?

File: examples/counter/iOS/project.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        UISupportedInterfaceOrientations:
+          - UIInterfaceOrientationPortrait
+          - UIInterfaceOrientationLandscapeLeft
+          - UIInterfaceOrientationLandscapeRight
+        UILaunchScreen: {}
+    settings:
+      OTHER_LDFLAGS: [-w]
+      SWIFT_OBJC_BRIDGING_HEADER: generated/sharedFFI.h
+    buildRules:
+      - name: Generate FFI
```
Comment (charypar) [inline examples/counter/iOS/project.yml:34]: Ohh, this is greate to have captured in human managed code.

**Review by charypar — APPROVED** [review]
Looks very promising! I only have that one question, no comments, looks good to me.

File: docs/src/getting_started/ios_with_xcodegen.md (hunk 1 above)
Comment (StuartHarris) [reply docs/src/getting_started/ios_with_xcodegen.md:83]: you can do either. Every time you run it the Xcode project file is regenerated, so either use it to get going and then use Xcode exclusively, or keep the `project.yml` file up to date and keep using that.

**Pull request merged**